parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -65,28 +68,25 @@
   private ParquetUtil() {
   }
 
-  // Access modifier is package-private, to only allow use from existing tests
-  static Metrics fileMetrics(InputFile file) {
-    return fileMetrics(file, MetricsConfig.getDefault());
-  }
-
   public static Metrics fileMetrics(InputFile file, MetricsConfig metricsConfig) {
     return fileMetrics(file, metricsConfig, null);
   }
 
   public static Metrics fileMetrics(InputFile file, MetricsConfig metricsConfig, NameMapping nameMapping) {
     try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(file))) {
-      return footerMetrics(reader.getFooter(), metricsConfig, nameMapping);
+      return footerMetrics(reader.getFooter(), Stream.empty(), null, metricsConfig, nameMapping);
     } catch (IOException e) {
       throw new RuntimeIOException(e, "Failed to read footer of file: %s", file);
     }
   }
 
-  public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metricsConfig) {
-    return footerMetrics(metadata, metricsConfig, null);
+  public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetrics> fieldMetrics,
+                                      Schema inputSchema, MetricsConfig metricsConfig) {
+    return footerMetrics(metadata, fieldMetrics, inputSchema, metricsConfig, null);
   }
 
-  public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metricsConfig, NameMapping nameMapping) {
+  public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetrics> fieldMetrics,
```

Comment:
Why we want to change the interface to include `inputSchema`? Line 99-100 below already creates the `fieldSchema`, are they different?

Reply:
I think this might be related to passing the Parquet field IDs. Since we don't need to do that, we can probably remove the new method here.

Reply:
Oh, I see below that this is used to convert a metrics ID to a field alias. That looks fine to me.